pantheon-bundle/src/main/java/com/redhat/pantheon/model/module/Module.java

```diff
@@ -45,125 +44,27 @@
     @Named("jcr:uuid")
     Field<String> uuid();
 
-    default Child<ModuleLocale> moduleLocale(Locale locale) {
+    default Child<ModuleLocale> locale(Locale locale) {
         return child(locale.toString(), ModuleLocale.class);
     }
 
     default Optional<ModuleVersion> getDraftVersion(@Nonnull final Locale locale,
                                                     @Nonnull final String variantName) {
-        return traverseFrom(this)
-                .toChild(m -> m.moduleLocale(locale))
-                .toChild(ModuleLocale::variants)
-                .toChild(variants -> variants.variant(variantName))
-                .toChild(ModuleVariant::draft)
-                .getAsOptional();
+        return (Optional<ModuleVersion>) Document.super.getDraftVersion(locale, variantName);
     }
 
     default Optional<ModuleVersion> getReleasedVersion(@Nonnull final Locale locale,
-                                                    @Nonnull final String variantName) {
-        return traverseFrom(this)
-                .toChild(m -> m.moduleLocale(locale))
-                .toChild(ModuleLocale::variants)
-                .toChild(variants -> variants.variant(variantName))
-                .toChild(ModuleVariant::released)
-                .getAsOptional();
+                                                       @Nonnull final String variantName) {
+        return (Optional<ModuleVersion>) Document.super.getReleasedVersion(locale, variantName);
```

Comment:
I guess we still need this because of generics, but it's much more succinct than before 👍